server.go

```diff
@@ -781,12 +781,7 @@ func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.
 	}
 	reply, appErr := md.Handler(srv.server, stream.Context(), df, s.opts.unaryInt)
 	if appErr != nil {
-		appStatus, ok := status.FromError(appErr)
-		if !ok {
-			// Convert appErr if it is not a grpc status error.
-			appErr = status.Error(convertCode(appErr), appErr.Error())
-			appStatus, _ = status.FromError(appErr)
-		}
+		appStatus, _ := status.FromError(toRPCErr(appErr))
```

Comment:
This is the application error, which is not what toRPCErr is supposed to handle.  This is where convertCode is appropriate to use.